framework/src/actions/Action.C

```diff
@@ -58,3 +58,9 @@ Action::getShortName() const
 {
   return _name.substr(_name.find_last_of('/') != std::string::npos ? _name.find_last_of('/') + 1 : 0);
 }
+
+std::string
+Action::getBaseName() const
+{
+  return _name.substr(0, _name.find_last_of('/') != std::string::npos ? _name.find_last_of('/') : 0);
```

Comment:
Two things: 

1. If there are multiple slashes, like ./foo/bar/baz, this will return ./foo/bar.  Is that correct, or did you want it to return ./foo?
1. If there are no slashes, this will return an empty string.  Is that correct?

If yes to both, would you mind documenting that this is the expected behavior of Action::getBaseName()?